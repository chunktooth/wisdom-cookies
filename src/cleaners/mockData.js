/* exported variableName */
export const mockQuotes = [
  {
    'message': 'Mankind is unkind, man',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeed!',
    'id': '1234'
  }
];