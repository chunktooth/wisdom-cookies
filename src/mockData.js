/* exported variableName */
export const mockWisdoms = [
  {
    'message': 'Mankind is unkind, man..',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeeeed!',
    'id': '1234'
  }
];

export const mockJar = [
  {
    'message': 'Mankind is unkind, man',
    'id': '5280'
  },
  {
    'message': 'Godluck and Goodspeed!',
    'id': '1234'
  }
];

export const mockMoonphases = { response: [
  { 
    timestamp: 1524433610, 
    dateTimeISO: "2018-04-22T21:46:50+00:00", 
    code: 1, 
    name: "first quarter"
  },
  {
    timestamp: 1525049954, 
    dateTimeISO: "2018-04-30T00:59:14+00:00", 
    code: 2, 
    name: "full moon"
  }]
};

export const mockMoonphasesArray = [{ response: [
  { 
    timestamp: 1524433610, 
    dateTimeISO: "2018-04-22T21:46:50+00:00", 
    code: 1, 
    name: "first quarter"
  },
  {
    timestamp: 1525049954, 
    dateTimeISO: "2018-04-30T00:59:14+00:00", 
    code: 2, 
    name: "full moon"
  }]
}]; 

export const mockMoonphase =  {
  date: "2018-04-22", 
  name: "first quarter"
};

export const mockWisdom = {
  'message': 'In Crust We Trust',
  'id': '1234'
};

export const mockCleanedMoons = [
  {
    date: "2018-04-22", 
    name: "first quarter"
  },
  { 
    date: "2018-04-30", 
    name: "full moon"
  }
];