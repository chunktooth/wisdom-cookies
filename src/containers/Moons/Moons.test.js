import React from 'react';
import { Moons,
  mapStateToProps, 
  mapDispatchToProps } from './Moons';
import { shallow } from 'enzyme';
import { mockMoonphasesArray } from '../../mockData';


describe('Moons', () => {
  let wrapper;
 
  beforeEach(() => {
    wrapper = shallow(<Moons 
      moonphases={mockMoonphasesArray} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map moonphases state to props', () => {
    const mappedMoonphases = mapStateToProps(mockMoonphasesArray);
    expect(mappedMoonphases.moonphases).toEqual(mockMoonphasesArray.moonphases);
  });

  it('should dispatch loadMoons to props', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.loadMoons();
    expect(mockDispatch).toHaveBeenCalled();
  });

});