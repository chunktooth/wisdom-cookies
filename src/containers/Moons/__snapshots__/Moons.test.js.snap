// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moons should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Moons
    moonphases={
        Array [
            Object {
              "response": Array [
                Object {
                  "code": 1,
                  "dateTimeISO": "2018-04-22T21:46:50+00:00",
                  "name": "first quarter",
                  "timestamp": 1524433610,
                },
                Object {
                  "code": 2,
                  "dateTimeISO": "2018-04-30T00:59:14+00:00",
                  "name": "full moon",
                  "timestamp": 1525049954,
                },
              ],
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="overhead"
>
          "Which moon pie will be in the sky...?"
</p>,
        <div
          className="show-moons"
>
          <Unknown
                    phase={
                              Object {
                                        "response": Array [
                                          Object {
                                            "code": 1,
                                            "dateTimeISO": "2018-04-22T21:46:50+00:00",
                                            "name": "first quarter",
                                            "timestamp": 1524433610,
                                          },
                                          Object {
                                            "code": 2,
                                            "dateTimeISO": "2018-04-30T00:59:14+00:00",
                                            "name": "full moon",
                                            "timestamp": 1525049954,
                                          },
                                        ],
                                      }
                    }
          />
</div>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          id="jar"
          to="/jar"
>
          <div
                    className="jar-o-wisdoms"
          >
                    <img
                              alt="Revisit a jar of wisdoms"
                              className="wisdom-jar"
                              src="wisdom-jar.png"
                    />
                    <p
                              className="btn-txt"
                    >
                              Back to Jar
                    </p>
          </div>
</NavLink>,
      ],
      "className": "Moons",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "\\"Which moon pie will be in the sky...?\\"",
          "className": "overhead",
        },
        "ref": null,
        "rendered": "\\"Which moon pie will be in the sky...?\\"",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              phase={
                            Object {
                                          "response": Array [
                                            Object {
                                              "code": 1,
                                              "dateTimeISO": "2018-04-22T21:46:50+00:00",
                                              "name": "first quarter",
                                              "timestamp": 1524433610,
                                            },
                                            Object {
                                              "code": 2,
                                              "dateTimeISO": "2018-04-30T00:59:14+00:00",
                                              "name": "full moon",
                                              "timestamp": 1525049954,
                                            },
                                          ],
                                        }
              }
/>,
          ],
          "className": "show-moons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "phase": Object {
                "response": Array [
                  Object {
                    "code": 1,
                    "dateTimeISO": "2018-04-22T21:46:50+00:00",
                    "name": "first quarter",
                    "timestamp": 1524433610,
                  },
                  Object {
                    "code": 2,
                    "dateTimeISO": "2018-04-30T00:59:14+00:00",
                    "name": "full moon",
                    "timestamp": 1525049954,
                  },
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <div
            className="jar-o-wisdoms"
>
            <img
                        alt="Revisit a jar of wisdoms"
                        className="wisdom-jar"
                        src="wisdom-jar.png"
            />
            <p
                        className="btn-txt"
            >
                        Back to Jar
            </p>
</div>,
          "id": "jar",
          "to": "/jar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Revisit a jar of wisdoms"
                className="wisdom-jar"
                src="wisdom-jar.png"
/>,
              <p
                className="btn-txt"
>
                Back to Jar
</p>,
            ],
            "className": "jar-o-wisdoms",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Revisit a jar of wisdoms",
                "className": "wisdom-jar",
                "src": "wisdom-jar.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Back to Jar",
                "className": "btn-txt",
              },
              "ref": null,
              "rendered": "Back to Jar",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="overhead"
>
            "Which moon pie will be in the sky...?"
</p>,
          <div
            className="show-moons"
>
            <Unknown
                        phase={
                                    Object {
                                                "response": Array [
                                                  Object {
                                                    "code": 1,
                                                    "dateTimeISO": "2018-04-22T21:46:50+00:00",
                                                    "name": "first quarter",
                                                    "timestamp": 1524433610,
                                                  },
                                                  Object {
                                                    "code": 2,
                                                    "dateTimeISO": "2018-04-30T00:59:14+00:00",
                                                    "name": "full moon",
                                                    "timestamp": 1525049954,
                                                  },
                                                ],
                                              }
                        }
            />
</div>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            id="jar"
            to="/jar"
>
            <div
                        className="jar-o-wisdoms"
            >
                        <img
                                    alt="Revisit a jar of wisdoms"
                                    className="wisdom-jar"
                                    src="wisdom-jar.png"
                        />
                        <p
                                    className="btn-txt"
                        >
                                    Back to Jar
                        </p>
            </div>
</NavLink>,
        ],
        "className": "Moons",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "\\"Which moon pie will be in the sky...?\\"",
            "className": "overhead",
          },
          "ref": null,
          "rendered": "\\"Which moon pie will be in the sky...?\\"",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                phase={
                                Object {
                                                "response": Array [
                                                  Object {
                                                    "code": 1,
                                                    "dateTimeISO": "2018-04-22T21:46:50+00:00",
                                                    "name": "first quarter",
                                                    "timestamp": 1524433610,
                                                  },
                                                  Object {
                                                    "code": 2,
                                                    "dateTimeISO": "2018-04-30T00:59:14+00:00",
                                                    "name": "full moon",
                                                    "timestamp": 1525049954,
                                                  },
                                                ],
                                              }
                }
/>,
            ],
            "className": "show-moons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "phase": Object {
                  "response": Array [
                    Object {
                      "code": 1,
                      "dateTimeISO": "2018-04-22T21:46:50+00:00",
                      "name": "first quarter",
                      "timestamp": 1524433610,
                    },
                    Object {
                      "code": 2,
                      "dateTimeISO": "2018-04-30T00:59:14+00:00",
                      "name": "full moon",
                      "timestamp": 1525049954,
                    },
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <div
              className="jar-o-wisdoms"
>
              <img
                            alt="Revisit a jar of wisdoms"
                            className="wisdom-jar"
                            src="wisdom-jar.png"
              />
              <p
                            className="btn-txt"
              >
                            Back to Jar
              </p>
</div>,
            "id": "jar",
            "to": "/jar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Revisit a jar of wisdoms"
                  className="wisdom-jar"
                  src="wisdom-jar.png"
/>,
                <p
                  className="btn-txt"
>
                  Back to Jar
</p>,
              ],
              "className": "jar-o-wisdoms",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Revisit a jar of wisdoms",
                  "className": "wisdom-jar",
                  "src": "wisdom-jar.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Back to Jar",
                  "className": "btn-txt",
                },
                "ref": null,
                "rendered": "Back to Jar",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
